mlir/lib/Conversion/AffineToStandard/AffineToStandard.cpp

```diff
@@ -334,7 +334,15 @@ class AffineYieldOpLowering : public OpRewritePattern<AffineYieldOp> {
 
   LogicalResult matchAndRewrite(AffineYieldOp op,
                                 PatternRewriter &rewriter) const override {
-    rewriter.replaceOpWithNewOp<scf::YieldOp>(op);
+    if (isa<scf::ParallelOp>(op.getParentOp())) {
+      // This could happen while lowering affine.parallel to scf.parallel.
+      // scf.parallel in all cases expect scf.yield to have no operands.
```

Comment:
Rewrite the last line for clarity:

`scf.parallel does not yield any values via its terminator scf.yield but models reductions differently using additional ops in its region.`